src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java

```diff
@@ -121,6 +126,11 @@ public OperationCompletionRS updateDashboard(ReportPortalUser.ProjectDetails pro
 	public OperationCompletionRS addWidget(Long dashboardId, ReportPortalUser.ProjectDetails projectDetails, AddWidgetRq rq,
 			ReportPortalUser user) {
 		Dashboard dashboard = getShareableDashboardHandler.getAdministrated(dashboardId, projectDetails);
+		BusinessRule.expect(dashboard.getDashboardWidgets().size() > MAX_WIDGET_ON_DASHBOARD, Predicates.equalTo(true)).verify(
```

Comment:
1. Pleas add test for this point.
2. Please, pay attention to branch naming - it must be with postfix.